README.md

```diff
@@ -1,23 +1,178 @@
+    "ruleLanguage": "Rego",
+    "match": [
+      {
+        "apiGroups": [
+          ""
+        ],
+        "apiVersions": [
+          "v1"
+        ],
+        "resources": [
+          "Pod"
+        ]
+      }
+    ],
+    "ruleDependencies": [
+    ],
+    "controlConfigInputs": [
+      {
+        "path": "settings.postureControlInputs.cpu_request_max",
+        "name": "cpu_request_max",
+        "description": "Ensure CPU max requests are set"
+      }
+    ],
+    "description": "CPU limits and requests are not set.",
+    "remediation": "Ensure CPU limits and requests are set.",
+    "ruleQuery": "armo_builtins"
+}
+```
+* Attribute "armoBuiltin": true - mandatory for armo rules. Only ARMO team members are authorized to create builtin objects.
+
+
+* See [rule go struct](https://github.com/kubescape/opa-utils/blob/fc41353af5c67221dead93c353b48e88f02a9fe4/reporthandling/datastructures.go#L38) for further explanations of rule fields
+* Optional attributes :
+  * "hostSensorRule": "true" - indicates that rule gets information from host scanner
+
+  * "useFromKubescapeVersion" - add if rule is only supported from a certain Kubescape version. Inclusive.
+
+  * "useUntilKubescapeVersion" - add if newer version exists so control doesn’t run both. Inclusive. 
+
+  * "imageScanRelated": true - indicates that rule uses information from image scanning
+
+  * "controlConfigInputs" - A list the rule uses and can be configured by the user. See example above.
+
+
+3. Add to rule directory file - raw.rego
+
+    This is where the logic of the rule is. 
+    Example of raw.rego:
+    ```
+    package armo_builtins
+
+    deny[msga] {
+
+        pod := input[_]
+        pod.kind == "Pod"
+        container := pod.spec.containers[i]
+        container.securityContext.privileged == true
+        path := sprintf("containers[%d].securityContext.privileged", [i])
+
+        msga := {
+            "alertMessage": sprintf("pod: %v is defined as privileged", [pod.metadata.name]),
+            "packagename": "armo_builtins",
+            "fixPaths": [],
+            "failedPaths": path,
+            "alertObject": {
+                "k8sApiObjects": [pod]
+            }
+        }
+    }
+    ```
+    Use [opa rego reference](https://www.openpolicyagent.org/docs/latest/policy-reference/) for help with syntax
+
+    See struct of a [rule response](https://github.com/kubescape/opa-utils/blob/d654a601023efd8a102c5bd43a4b9115ebf692c4/reporthandling/datastructuresv1.go#L23)
+
+
+4. Add a test for new rule (and run it!), see how [here](/testrunner/README.md)
+5. Add filter.rego if needed - If exists, the filter is run by kubescape to calculate ‘all resources’ = the number of potential resources to fail. Affects the risk score. Needed in cases where rule asks for resources that are not potential to fail, eg- if a rule asks for pods and service accounts to see if they are connected but only fails the pods, we would create a filter rego that returns only pods.
```

Comment:
```suggestion
5. Add `filter.rego` if needed - If exists, the filter is run by Kubescape to calculate ‘all resources’ = the number of potential resources to fail. It affects the risk score. Needed in cases where rule asks for resources that are not potential to fail, eg- if a rule asks for pods and service accounts to see if they are connected but only fails the pods, we would create a filter rego that returns only pods.
```
Probably worth explaining how a rule or filter can 'return' the pods (meaning, through the `k8sObjects`, etc).
I also would like to see here an explanation of the whole `msg` structure

Reply:
`msg` structure is referenced here : See struct of a [rule response](https://github.com/kubescape/opa-utils/blob/master/reporthandling/datastructuresv1.go#L23)